db/crud.go

```diff
@@ -592,6 +592,13 @@ func (db *Database) ImportDoc(docid string, value []byte, isDelete bool) error {
 	}
 
 	_, err := db.updateDoc(docid, true, 0, func(doc *document) (Body, AttachmentData, error) {
+		// If this is a delete, and there is no xattr on the existing doc,
+		// we shouldn't import.  (SG purge arriving over DCP feed)
+		if isDelete && doc.CurrentRev == "" {
+			base.LogTo("Import+", "Import skipped for delete of doc with no existing SG xattr (SG purge): %s", docid)
+			return nil, nil, base.ErrImportCancelled
```

Comment:
I'm using ErrImportCancelled to cover any scenarios where the import is intentionally cancelled (i.e. not error scenarios).  Right now that's just the purge case, there will be others in future once we add the mobile opt-in capability.  I think it makes sense to reuse the error code in both cases, instead of having a list of functionally equivalent error codes on line 621 below.  

However, I think the verbose logging should provide the details on the exact scenario, which was the goal here.  Personally I thought that 'import skipped' seemed less like an error than 'import cancelled', and is clearer to read than 'Import ignored for...', but if it's not clear to you I'll try to rephrase it.